tests/integration_tests/performance/test_net_tput.py

```diff
@@ -0,0 +1,330 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Tests the network throughput overhead added by Firecracker."""
+
+import json
+import time
+import statistics
+import socket
+from subprocess import run, PIPE
+import pytest
+
+from retry import retry
+
+import host_tools.network as net_tools
+from framework.statistics import Statistics
+
+
+RUNS = 1
+
+IPERF = 'iperf3'
+
+# Iteration duration of a single run in seconds.
+ITERATION_DURATION = 1
+
+WINDOW_SIZE_MB = 10
+
+# BW in Mbits.
+BANDWIDTHS = {
+    'udp': [100, 5000, 10000],
+    'tcp': [5000, 10000, 20000, 0]
+}
+
+# Packet size used for TCP.
+PACKET_SIZE_KB = [128, 256, 1024, 1500]
+
+# Flow direction and corresponding modes for iperf.
+MODES = {
+    'h2g': '-R',
+    'g2h': ''
+}
+
+# Options used in iperf for each protocol.
+PROTOCOL = {
+    'tcp': '',
+    'udp': '-u'
+}
+
+# Skip the first OMIT_TIME seconds when testing TCP
+# to avoid slow start.
+OMIT_TIME = 0
+
+# Pin iperf on this CPU.
+PINNING_CPU = 1
+
+OUTPUT_JSON = 'test_net_tput_{}_iperfs.json'
+
+CSET_NAME = 'test_tput'
+
+class IperfClient:
+    def __init__(self, protocol, host_ip, iperfs, microvm):
+        assert protocol == 'udp' or protocol == 'tcp'
+        self.protocol = protocol
+        self.host_ip = host_ip
+        self.iperfs = iperfs
+        self.microvm = microvm
+
+    def run_iperf_client(self):
```

Comment:
In `IperfClient` the `run_iperf_client` method returns a map of parameters. In `IperfTcpClient` and `IperfUdpClient` it actually runs the iperf command. Also this methods take different parameters. This is a bit confusing. I would have `run_iperf_client` run the iperf command, as the name suggests, and have a different method for building the params map.